flink-runtime/src/main/java/org/apache/flink/runtime/taskexecutor/slot/TaskSlotTableImpl.java

```diff
@@ -200,6 +200,17 @@ public boolean isClosed() {
 		}
 	}
 
+	@Override
+	public Set<AllocationID> getActiveTaskAllocationIdsPerJob(JobID jobId) {
+		Iterator<TaskSlot<T>> taskSlotIterator = new TaskSlotIterator(jobId, TaskSlotState.ACTIVE);
+		Set<AllocationID> allocationIds = new HashSet<>();
+		while (taskSlotIterator.hasNext()) {
+			allocationIds.add(taskSlotIterator.next().getAllocationId());
+		}
+
+		return allocationIds;
```

Comment:
```suggestion
		return Sets.newHashSet(new TaskSlotIterator(jobId, TaskSlotState.ACTIVE);
```

might be a bit shorter.

Reply:
```suggestion
		return Sets.newHashSet(new TaskSlotIterator(jobId, TaskSlotState.ACTIVE))
			.stream()
			.map(TaskSlot::getAllocationId)
			.collect(Collectors.toSet());
```
The only other "shorter" way I can imagine is the one above. I just hesitated because it would mean iterating over the collection twice just to save a few lines.

Reply:
Yes, you are completely right. Forget about my comment. I overlooked that we are only interested in the `AllocationIDs`.